src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -43,13 +66,29 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
     }
 
     execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const { payment, ...order } = payload;
+        const { payment } = payload;
 
         if (!payment || !payment.methodId) {
-            throw new InvalidArgumentError('Unable to submit payment because "payload.payment.methodId" argument is not provided.');
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const paymentName = payment.methodId;
+        const { methodId } = payment;
+
+        return this._getPaymentData(payment)
+            .then(paymentData => {
+                const paymentPayload = { methodId, paymentData };
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder(payload, options))
```

Comment:
I think for SquareV2, you should exclude payment data from the order payload, and submit payment data directly to our payment service instead. Can you change back to how it was before? Thanks!

Reply:
@davidchin you mean using bigpay client instead of `orderActionCreator`?

Reply:
No, what I meant was, what we had was good, the change you made broke the flow. If you look closely, you should see that previously we used `PaymentActionCreator` for submitting payment data.

Reply:
Hey @davidchin , I just pulled latests bits from master and it doesn't work either.

![image](https://user-images.githubusercontent.com/8570490/44559348-5acba680-a6fe-11e8-9e34-70278f450f56.png)


Please confirm this.